### PR TITLE
Add type definition pdf2image

### PR DESCRIPTION
**Pull request by taoqf**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/pdf2image/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export type OutputFormat = ((page_num: number, page_index: number, totalPagesProcessed: number, totalPDFPages: number, name: string, path: string, vm: typeof import ('vm')) => string) | string;
+
+export interface Options {
+	density: number;
+	height: number;
+	width: number;
+	outputType: 'jpg' | 'png' | '.jpg' | '.png';
+	quality: number;
+	pages: '*' | string;	// * | even | odd | '/1,/3,5-6,-8, 9-'
```
Comment (armanio123) [inline types/pdf2image/index.d.ts]: Remove `'*'` from the type as it's already defined in `string`.

File: types/pdf2image/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * T - the total of processed pages
+ * s - the name of the pdf file
+ * p - the path of the pdf file
+ * % - the character '%'
+ * {...} - a custom piece of code where all of the above values can be used
+ */
+
+/// <reference types="node" />
+
+export type OutputFormat = ((page_num: number, page_index: number, totalPagesProcessed: number, totalPDFPages: number, name: string, path: string, vm: typeof import ('vm')) => string) | string;
```
Comment (armanio123) [inline types/pdf2image/index.d.ts]: `page_num`, and `page_index` name is incorrect, should be `pageNum`, `pageIndex`  based on the documentation.

File: types/pdf2image/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export function convertPDF(pdfFilePath: string, options?: Partial<Options>): Promise<ConvertedFile[]>;
+
+export interface Converter {
+	convertPDF(pdfFilePath: string): Promise<ConvertedFile[]>;
+	convertPDFList(pdfList: string[]): Promise<ConvertedFile[]>;
+}
+
+export function compileConverter(options?: Partial<Options>): Converter;
```
Comment (armanio123) [inline types/pdf2image/index.d.ts]: Alternatively to `Partial<Options>` you can make all of the properties you like optional by appending a `?` at the end of the property name. i.e `density?: number`.

No change needed here just a suggestion.

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/pdf2image/index.d.ts (hunk 3 above)
Comment (taoqf) [reply types/pdf2image/index.d.ts]: Thanks.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**